# CIS 443-01 - Analytics Programming
## Program 1
__Due__: Monday September 23 (by 11:59 PM)<br />
__Worth__: 50 pts.<br />
__Purpose__: Use a function and repetition control structure to demonstrate compounding interest.

### Directions
In the code cell below, write Python code to accomplish the following using what you've learned through Chapter 4. 


#### Part 1

Your code should allow the user to enter three values: a principle balance, an interest rate, and a number of years. 

For simplicity's sake, we will compound our interest once per year. In order to do so, create a function interest_calculator with three inputs, corresponding to the user inputs. They should have the default values 10000, .09, and 10, respectively. 

Your formula to calculate the interest will be Balance(Year + 1) = Balance * (1 + interest rate)

So for example, starting with 10,000 and a 5% interest rate, the formula is 10,000 * (1 + .05) = 10,500. The next year would be 10,500 * (1 + .05) = 11,025 and so on. 

Each year you will output the year number, and the value of the balance that year. 


In [66]:
# Put Program 1 Code Here. Do not create any additional code blocks or points may be deducted. 
#CIS-443-01
#Program 1
#9/23/24
#This program takes a balance, interest rate, and time in years to calculate compounded interest for every year

#Calls function using principal, interest_rate, and years.
def compounding_interest_calculator(principal, interest_rate, years):
    current = 1
    #The while loop will continue until it reaches the year the user inputs
    while current <= years:

                #calculates the equation, prints the values, and steps up the current year after
                principal = principal * (1 + interest_rate)
                print(f"Year {current} compounded value is ${principal:.2f}")
                current += 1

#Princable balance input
pbal = input("Enter the principle balance: ")

#Checks principle balance for a float value
try:
    pbal = float(pbal)
    #Interest input
    inrate = input("Enter the interest rate: ")

    #Checks interest for a float value
    try:
        inrate = float(inrate)
        #time in years input
        years = input("Enter the number of years: ")

        #checks years for an integer 
        try:
            years = int(years)

            #calls the defined function
            compounding_interest_calculator(pbal, inrate, years)
            
        except ValueError:
            print(years, "is not valid.")
    
    except ValueError:
            print(inrate, "is not valid.")
        
except ValueError:
    print(pbal, "is not valid.")


Enter the principle balance:  14339
Enter the interest rate:  .11
Enter the number of years:  10


Year 1 compounded value is $15916.29
Year 2 compounded value is $17667.08
Year 3 compounded value is $19610.46
Year 4 compounded value is $21767.61
Year 5 compounded value is $24162.05
Year 6 compounded value is $26819.87
Year 7 compounded value is $29770.06
Year 8 compounded value is $33044.77
Year 9 compounded value is $36679.69
Year 10 compounded value is $40714.46


### Example Output
<div class="alert alert-block alert-info">
Enter the principle balance:  14339<br>
Enter the interest rate:  .11<br>
Enter the number of years:  10<br>
Year 1 compounded value is \$15916.29<br>
Year 2 compounded value is \$17667.08<br>
Year 3 compounded value is \$19610.46<br>
Year 4 compounded value is \$21767.61<br>
Year 5 compounded value is \$24162.05<br>
Year 6 compounded value is \$26819.87<br>
Year 7 compounded value is \$29770.06<br>
Year 8 compounded value is \$33044.77<br>
Year 9 compounded value is \$36679.69<br>
Year 10 compounded value is \$40714.46
</div>

#### Part 2
**Answer the following**:

How many years of this calculation do you have to use before you first get 38,000, at 5% interest, starting with 7,500?<br />
34<br />
100,000?<br />
54<br />
1,000,000?<br />
101<br />

Double click the cell in Jupyter to edit (__5 pts.__)



Be sure to save and exit your Jupyter Notebook and Shutdown Jupyter Lab (from the __File__ menu) before you submit your notebook on Blackboard for grading.